rest/route/host_agent.go

```diff
@@ -245,7 +245,6 @@ func (h *hostAgentNextTask) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	setNextTask(nextTask, &nextTaskResponse)
-	nextTaskResponse.UnsetFunctionVarsDisabled = flags.UnsetFunctionVarsDisabled
```

Comment:
Removing this in the same change as the agent change will temporarily cause a reversion in behavior for old agents still on version 2023-11-02, and those agents will stop unsetting function vars until the new version is deployed. If `UnsetFunctionVarsDisabled` is not set with the next task request anymore (i.e. the app server updates) but the old agent is still running (e.g. because it's running a task group), the 2023-11-02 agent will interpret this to mean it should keep the function vars (i.e. it will not unset them when the function ends). This could potentially break someone's tasks if they've been updated to depend on the new behavior (i.e. unset function vars) and it temporarily reverts back to keeping function vars.

If we want to do this more safely without the temporary reversion to not unsetting function vars, then we have to roll out the agent change first (to always unset function vars) and then delete the admin setting in a separate commit once the agents have all deployed.

Reply:
good catch, thank you! 
opened https://github.com/evergreen-ci/evergreen/pull/7206